src/openrct2/cmdline/BenchUpdate.cpp

```diff
@@ -0,0 +1,160 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2020 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "CommandLine.hpp"
+
+#ifdef USE_BENCHMARK
+
+#    include "../Context.h"
+#    include "../GameState.h"
+#    include "../OpenRCT2.h"
+#    include "../platform/Platform2.h"
+#    include "../platform/platform.h"
+
+#    include <benchmark/benchmark.h>
+#    include <cstdint>
+#    include <iterator>
+#    include <vector>
+
+using namespace OpenRCT2;
+
+static void BM_update(benchmark::State& state, const std::string& filename)
+{
+    std::unique_ptr<IContext> context(CreateContext());
+    if (context->Initialise())
+    {
+        if (!filename.empty() && !context->LoadParkFromFile(filename))
```

Comment:
What if the filename is empty? The function will just run, is that expected?

Reply:
It will fail to load in `LoadParkFromFile` and get skipped. I believe that's fine. Additionally, as this is an explicitly provided parameter by a user to the `benchsimulate` command, I think the user will be aware of what to do with such problem.

Reply:
Ok